keps/sig-apps/5547-workload-ref-in-job/README.md

```diff
@@ -0,0 +1,401 @@
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- `pkg/controller/job/job_controller_test.go`: `09-30-2025` - `89.1%`
+- `pkg/registry/batch/job/strategy_test.go`: `09-30-2025` - `93%`
+- `pkg/apis/batch/validation/validation_test.go`: `09-30-2025` - `86.3%`
+
+##### Integration tests
+
+Update the following test cases to include the option of workloadRef field:
+
+- [TestIndexedJob](https://github.com/kubernetes/kubernetes/blob/master/test/integration/job/job_test.go#L2905): [result]()
+- [TestImmediateJobRecreation](https://github.com/kubernetes/kubernetes/blob/master/test/integration/job/job_test.go#L2291) : [result]()
+- [TestParallelJob](https://github.com/kubernetes/kubernetes/blob/master/test/integration/job/job_test.go#L2671) : [result]()
+- [BenchmarkLargeIndexedJob](https://github.com/kubernetes/kubernetes/blob/master/test/integration/job/job_test.go#L3508) : [result]()
+- [TestJobFailedWithInterrupts](https://github.com/kubernetes/kubernetes/blob/master/test/integration/job/job_test.go#L3908): [result]()
+
+
+##### e2e tests
+
+Add the following new e2e test case:
+
+- [should run a job to completion when workload reference is added](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/job.go): [result]()
+
+### Graduation Criteria
+
+#### Alpha
+
+- Job controller populates new field via Same Gang Scheduler FeatureGate.
+- Unit and integration tests passed as designed in [TestPlan](#test-plan).
+
+#### Beta
+
+- E2E tests passed as designed in [TestPlan](#test-plan).
+- Feature is enabled by default.
+- Address all issues reported by users.
+
+#### GA
+
+- No negative feedback.
+
+### Upgrade / Downgrade Strategy
+
+- Upgrade
+  - If the feature gate is enabled, `Job` workloads are allowed to use only.
+  - Even if the feature gate is enabled, but the `Job` workloads don't have the `workloadRef` field, the Job controller should function normally.
+- Downgrade
+  - Previously configured values will be ignored, and the job will be running normally but pods won't scheduled all at once, instead pod by pod.
+
+### Version Skew Strategy
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: Workload/GenericWorkload/NativeWorkload
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kube-scheduler
+  - Feature gate name: GangScheduling
+  - Components depending on the feature gate:
+    - kube-scheduler
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
```

Comment:
Hmmm... I don't think re-using the same feature gate as [KEP-4671](https://github.com/kubernetes/enhancements/pull/5558) is advisable. I'd expect separate one just for that field. You can express that this FG relies on the other ones, we've done that in the past (see most of DRA).